### PR TITLE
Improve all documentation for tomland and fix all Haddock warnings

### PR DESCRIPTION
**Pull request by ChShersh**
Just writing great documentation in general :slightly_smiling_face:

### CODE REVIEW

File: src/Toml/Bi/Monad.hs [hunk 1]
```diff
@@ -22,7 +22,7 @@ import Data.Coerce (Coercible, coerce)
 
 In practice instead of @r@ we will use some @Reader Toml@ and instead of @w@ we will
 use @State Toml@. This approach with the bunch of utility functions allows to
-have single description for from/to 'Toml' conversion.
+have single description for from/to @TOML@ conversion.
```
Comment (willbasky) [inline src/Toml/Bi/Monad.hs]: Here, it is emphasized, but above is not.

File: src/Toml/Printer.hs [hunk 2]
```diff
@@ -58,18 +63,17 @@ will be translated to this
 title = "TOML Example"
 
 [example.owner]
-  name = "Kowainik"
-
+  name = \"Kowainik\"
```
Comment (willbasky) [inline src/Toml/Printer.hs]: Does it not work without `\`?

**Comment by willbasky** [comment]
Looks clearer now :+1:

File: src/Toml/Bi/Monad.hs (hunk 1 above)
Comment (ChShersh) [reply src/Toml/Bi/Monad.hs]: I used `@` in some places but then got tired... This doesn't really matter much, just some formatting :slightly_smiling_face:

File: src/Toml.hs [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
+    , userAge  :: Int
+    }
+
+userCodec :: 'TomlCodec' User
+userCodec = User
+    \<$\> Toml.'text' "name" '.=' userName
+    \<*\> Toml.'int'  "age"  '.=' userAge
+@
+
+For bigger examples see README.md in the repository:
```
Comment (vrom911) [inline src/Toml.hs]:
```suggestion
For more detailed examples see README.md in the repository:
```

File: src/Toml.hs [hunk 4]
```diff
@@ -1,9 +1,32 @@
...
-suggested to import this module qualified, like this:
+recommended to import this module qualified, like this:
 
 @
-import qualified Toml
+__import__ Toml (TomlCodec, (.=))
+__import__ __qualified__ Toml
 @
+
+Simple @'TomlCodec'@ looks like this:
```
Comment (vrom911) [inline src/Toml.hs]:
```suggestion
Simple @'TomlCodec'@ could be written in the following way:
```

**Pull request merged**